app/assets/javascripts/host_edit.js

```diff
@@ -295,10 +295,7 @@ function serializeForm() {
 
 //This function decides for a given input element is it set explicitly by the user.
 function isExplicit(index, element) {
-  if (element.nextSibling == undefined) return true;
-  if (element.nextSibling.children == undefined) return true;
-  if (element.nextSibling.children[0] == undefined) return true;
-  if (element.nextSibling.children[0].dataset == undefined) return true;
+  if (!element.classList.contains("has-explicit")) return true;
```

Comment:
`Explicit` is not the same as `disabled`:
Example: 
Host without hostgroup will have its fields enabled (environment_id for instance) but when I set the hostgroup, I don't want to send its value to the server since the user did not cancel the inheritance explicitly.